src/persist-client/src/batch.rs

```diff
@@ -799,31 +797,56 @@ where
 
         match self.buffer.push(key, val, ts.clone(), diff.clone()) {
             Some(part_to_flush) => {
-                self.flush_part(stats_schemas, part_to_flush).await;
+                self.flush_part(part_to_flush).await;
                 Ok(Added::RecordAndParts)
             }
             None => Ok(Added::Record),
         }
     }
 
+    /// Adds a batch of updates all at once. The caller takes responsibility for ensuring
+    /// that the data is appropriately chunked.
+    ///
+    /// The update timestamps must be greater or equal to `lower` that was given
+    /// when creating this [BatchBuilder].
+    pub async fn add_many(&mut self, updates: BlobTraceUpdates) -> Result<(), InvalidUsage<T>> {
+        for ts in updates.records().timestamps().iter().flatten() {
+            let ts = T::decode(ts.to_le_bytes());
+            if !self.lower.less_equal(&ts) {
+                return Err(InvalidUsage::UpdateNotBeyondLower {
+                    ts,
+                    lower: self.lower.clone(),
+                });
+            }
+
+            self.inclusive_upper.insert(Reverse(ts));
+        }
+
+        // If there have been any individual updates added, flush those first.
+        // This is a noop if there are no such updates.
+        // TODO: consider moving the individual updates to BatchBuilder?
+        let previous = self.buffer.drain();
+        self.flush_part(previous).await;
```

Comment:
What's the rationale for always flushing the previous parts? It seems like it might be nice to flush the current updates with the previous into a single blob?

Reply:
> What's the rationale for always flushing the previous parts?

Simplest thing to do, basically... we only hit this case if someone was mixing the two types of calls, which never happens in practice.

> It seems like it might be nice to flush the current updates with the previous into a single blob?

There's some risk that it outputs a blob that's larger than our target size. (So we could make it conditional, but for the above reason I'm not inclined to put much effort in!.)